src/ocean/Assets.ts

```diff
@@ -239,82 +239,96 @@ export class Assets extends Instantiable {
     return (await this.ocean.metadatacache.queryMetadata(searchQuery)).results
   }
 
+  /**    Metadata updates
+   *  Don't forget to call ocean.OnChainMetadataCache.update after using this functions
```

Comment:
and why does this method here not do that for me?

Reply:
got it, because of the method split-up. I know conceptually updating timeout is not updating the metadata service but maybe it makes sense to handle that within `editMetadata` too is since from user perspective it's also just metadata.

So `EditableMetadata` could have:

```ts
timeout?: {
  serviceIndex: number // could also ask for `serviceName` which would be a bit more user friendly
  value: number
}
```

and if passed we handle that in that method (where `updateServiceTimeout` can be left as individual method, but called here), allowing us to also call `ocean.OnChainMetadataCache.update` at the end. Thoughts?

Reply:
I'm afraid that we are going to end up with a huge function that does all..  also, don't forget that we have compute updates as well, and who knows what kind of updates are we going to have in the future..
That's why I thought that it makes sense to have update helpers (functions that are going to update parts of the ddo) and once you are done, you only have 1 call that updates the ddo on chain.
What if you want to change computePrivacy, timeout and asset's name ?

Reply:
guess I'm arguing for a helper method doing all then, calling the sub-functions. Since we know which fields are editable we should provide a helper so devs do not need to figure out for themselves which parts of DDO are editable and then which ones require which method to update.

This helper method handling all editable fields could simply call the individual sub-functions. Then it also stays flexible, advanced users can use the individual methods, but most will use the main helper method with a bunch of optional params. And then also stay consistent in the naming to group the methods, like:

- `editMetadata()`
  - `editMetadataFields()`
  - `editTimeout()`
  - `editComputePrivacy()`

If we decide to keep as is then at least solve this mixup of `edit` & `update` and stick to one